task_health_service.go

```diff
@@ -0,0 +1,65 @@
+package influxdb
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+// FindTasksService implements the FindTasks method of a TaskService for the TaskHealthService to use
+type FindTasksService interface {
+	FindTasks(ctx context.Context, filter TaskFilter) ([]*Task, int, error)
+}
+
+// TaskHealthService is responsible for monitoring the latency of active tasks in the kv store
+type TaskHealthService struct {
+	fts           FindTasksService
+	pollingPeriod time.Duration
+	ctx           context.Context
+	cancel        context.CancelFunc
+}
+
+// NewTaskHealthService takes a TaskHealthService and creates a new TaskHealthService
+func NewTaskHealthService(ctx context.Context, fts FindTasksService, p time.Duration) *TaskHealthService {
```

Comment:
By limiting this to generating metrics, we would no longer require the `ctx` or `p` arguments